SimTKcommon/include/SimTKcommon/internal/Xml.h

```diff
@@ -1708,10 +1709,76 @@ template <class T> inline
 auto fromXmlElementHelper(T& thing, Xml::Element& elt, 
                           const std::string& requiredTag, int) 
     -> void {
+    // TODO std::cout << "DEBUGfromXmlfree" << elt << std::endl;
     fromXmlElement(thing, elt, requiredTag); // call free function
 }
 
 
+inline void fromXmlElementHelperHelperRecurse(Xml::element_iterator eit,
+        Xml::element_iterator end) {
+    assert(eit == end);
+}
+
+/*
+// TODO decide between this and the other interface. 
+// With this interface, the call
+// is more verbose but also more clear:
+//
+// fromXmlElementHelperHelper("Foo", 1, e, "reqName",
+//     std::make_pair{bar, "bar"},
+//     std::make_pair{baz, "baz"}
+//     );
+template <typename T, typename ... TArgs> inline
+void fromXmlElementHelperHelperRecurse(Xml::element_iterator eit,
+        std::pair<T&, std::string> firstField,
+        std::pair<TArgs&, std::string> ... fields) {
```

Comment:
Hmmm.. I tried to get it to work again and I ran into some issues. It's still possible but it doesn't look as nice as I had hoped:


```
fromXmlElementHelperHelper("Foo", 1, e, "reqName",
    std::make_pair(std::ref(bar), "bar"),
    std::make_pair(std::ref(baz), "baz"));
```

The issue is that `make_pair` takes its arguments by value.